mcs/class/System/Mono.Btls/MonoBtlsProvider.cs

```diff
@@ -201,21 +201,80 @@ internal static bool ValidateCertificate (MonoBtlsX509Chain chain, MonoBtlsX509V
 			}
 		}
 
+		internal static void SetupCertificateStore (MonoBtlsX509Store store, MonoTlsSettings settings, bool server)
+		{
+#if MONODROID
+				SetupCertificateStore (store);
+				return;
+#else
+
+			if (settings == null || settings.CertificateSearchPaths == null) {
+				SetupCertificateStore (store);
+				return;
+			}
+
+			foreach (var path in settings.CertificateSearchPaths) {
+				if (string.Equals (path, "@default", StringComparison.Ordinal)) {
+					AddTrustedRoots (store, settings, server);
+					AddUserStore (store);
+					AddMachineStore (store);
+				} else if (string.Equals (path, "@user", StringComparison.Ordinal))
+					AddUserStore (store);
+				else if (string.Equals (path, "@machine", StringComparison.Ordinal))
+					AddMachineStore (store);
+				else if (string.Equals (path, "@trusted", StringComparison.Ordinal))
+					AddTrustedRoots (store, settings, server);
+				else if (path.StartsWith ("@pem:", StringComparison.Ordinal)) {
+					var realPath = path.Substring (5);
+					if (Directory.Exists (realPath))
+						store.AddDirectoryLookup (realPath, MonoBtlsX509FileType.PEM);
+				} else if (path.StartsWith ("@der:", StringComparison.Ordinal)) {
+					var realPath = path.Substring (5);
+					if (Directory.Exists (realPath))
+						store.AddDirectoryLookup (realPath, MonoBtlsX509FileType.ASN1);
+				} else {
+					if (Directory.Exists (path))
```

Comment:
The problem with using the enum is that we'd then need to store this as a list / array of `Pair< CertificateSearchPathType,string>` where the string only has a meaning for some of the enum values (Pem and Der).

We may actually get rid of the directory check since the native code might already do this, though I'd have to check to be sure.  However, we could certainly move the check into `AddCertificateSearchPath ()`.